### PR TITLE
Power Delivery Device

### LINKED ISSUE #45: Implementation of Power Device
### Scenario/Intent

Implement power device resource

### PR DESCRIPTION
**Pull request by hdiomede**
### Description

Implements power delivery device resource
### Issues Resolved
#45
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: examples/power_device.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+end
+
+# Removes an iPDU by hostname
+oneview_power_device '127.0.0.1' do
+  client my_client
+  action :remove
+end
+
+# Removes an iPDU using the name
+oneview_power_device '127.0.0.1 ' do
```
Comment (tmiotto) [inline examples/power_device.rb]: This is the same as the example before, something missing here maybe?

File: resources/power_device.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  add_if_missing
+end
+
+action :discover do
+  load_sdk
+  c = build_client(client)
+  power_devices_list = OneviewSDK::PowerDevice.get_ipdu_devices(c, name)
+  if power_devices_list.empty?
+    Chef::Log.info "#{resource_name} '#{name}' discovered."
+    converge_by "Adding #{resource_name} '#{name}'" do
```
Comment (tmiotto) [inline resources/power_device.rb]: `Adding` or `Discovering`?

File: resources/power_device.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      OneviewSDK::PowerDevice.discover(c, hostname: name, username: username, password: password)
+    end
+  else
+    Chef::Log.info("#{resource_name} '#{name}' is up to date")
+  end
+end
+
+action :remove do
+  unless remove
```
Comment (tmiotto) [inline resources/power_device.rb:49]: Where this `remove` is coming from, and why is it here?

**Review by tmiotto — CHANGES_REQUESTED** [review]

File: resources/power_device.rb (hunk 3 above)
Comment (hdiomede) [reply resources/power_device.rb:49]: Remove is from the resource base, it returns true or false.
It searches as a standard power device if the remove fails then we search as it as an iPDU

Comment (tmiotto) [reply resources/power_device.rb:49]: Okay, got it. I would just add a comment there explaining the workflow

**Review by tmiotto — APPROVED** [review]

**Pull request merged**